include/dlaf/eigensolver/reduction_to_band/mc.h

```diff
@@ -792,8 +792,8 @@ std::vector<hpx::shared_future<common::internal::vector<T>>> ReductionToBand<
   const auto ex_mpi = getMPIExecutor<Backend::MC>();
 
   common::Pipeline<comm::Communicator> mpi_col_chain_panel(grid.colCommunicator().clone());
-  common::Pipeline<comm::Communicator> mpi_row_chain(grid.rowCommunicator());
-  common::Pipeline<comm::Communicator> mpi_col_chain(grid.colCommunicator());
+  common::Pipeline<comm::Communicator> mpi_row_chain(grid.rowCommunicator().clone());
+  common::Pipeline<comm::Communicator> mpi_col_chain(grid.colCommunicator().clone());
```

Comment:
I'm just curious, is there a case where the communicator should *not* be cloned before sticking it in a pipeline?

Reply:
It's a good point!

The job of the `Pipeline` is to make the access to a resource serialized to ensure order. Ensuring this across different algorithms would require to move out the Pipeline from the algorithm (like we briefly discussed in the meeting, e.g. by having them in the `Communicator`) so the "continuity" of the order can be maintained.

The quick solution we opted for is to `clone` (`MPI_Comm_dup`) in order to get completely separate `Communicators`, which by construction do not "collide" across different algorithms.

So, the answer I would say is "yes, it always gets a clone". On the other hand, at the moment I cannot think of other quick/nice solutions (? embedding the clone in the `Pipeline<Communicator>` specialization).

Reply:
Yeah, putting the clone inside the construction what I was going for *if* it's always going to be cloned. But having it explicit is also fine by me. Since it'll hopefully be changed soon I don't mind leaving it like this.